server/ai_mediaserver.go

```diff
@@ -621,14 +622,144 @@ func (ls *LivepeerServer) StartLiveVideo() http.Handler {
 			GatewayRequestId: &requestID,
 			StreamId:         &streamID,
 		}
-		_, err = processAIRequest(ctx, params, req)
-		if err != nil {
-			stopPipeline(err)
+
+		params.node.LiveMu.Lock()
+		params.node.LivePipelines[params.liveParams.stream] = &core.LivePipeline{
+			RequestID: params.liveParams.requestID,
 		}
-		close(orchSelection)
+		params.node.LiveMu.Unlock()
+		close(pipelineStarted)
+
+		go processStream(ctx, params, req)
 	})
 }
 
+func processStream(ctx context.Context, params aiRequestParams, req worker.GenLiveVideoToVideoJSONRequestBody) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	orchSwapper := NewOrchestratorSwapper(params)
+
+	params.liveParams.outputWriter = startOutput(ctx, params)
+	for {
+		perOrchCtx, perOrchCancel := context.WithCancel(ctx)
+		params.liveParams = newParams(params.liveParams)
+
+		resp, err := processAIRequest(ctx, params, req)
+		if err != nil {
+			clog.Errorf(perOrchCtx, "Error processing AI Request: %s", err)
+			params.liveParams.stopPipeline(err)
+			perOrchCancel()
+			return
+		}
+
+		if err := startProcessing(perOrchCtx, params, resp); err != nil {
+			clog.Errorf(ctx, "Error starting processing: %s", err)
+			params.liveParams.stopPipeline(err)
+			perOrchCancel()
+			return
+		}
+
+		<-params.liveParams.processing
+		perOrchCancel()
+
+		if !orchSwapper.shouldSwap(ctx) {
+			break
+		}
+		clog.Infof(ctx, "Retrying stream with a different orchestrator")
+	}
+	params.liveParams.stopPipeline(fmt.Errorf("Done processing"))
```

Comment:
Should this be a `nil` error? Otherwise what would be the "happy exit" from this function? 🤔

Reply:
IMO it is alright to return errors for "happy" paths for the time being (we do it in a number of other places); it gives us somewhat granular sequencing without having to ask @ecmulli to handle new types of events.